apps/cpp_rtvm/main.cc

```diff
@@ -350,43 +350,46 @@ int ExecuteModel(ToolArgs& args) {
   } else if (!args.input.empty() && !args.output.empty()) {
     LOG(INFO) << "Executing with Input:" << args.input << " Output:" << args.output;
     // Set Input from Numpy Input
-    runner->SetInput(args.input);
+    runner.SetInput(args.input);
     // Run the model
-    runner->Run();
+    runner.Run();
     // Get Output as Numpy dump
-    runner->GetOutput(args.output);
+    runner.GetOutput(args.output);
   } else {
     LOG(INFO) << "Executing dry run ... ";
     // Set random input for all inputs
     for (auto& elem : mInfo.input_info) {
       LOG(INFO) << "Set Random Input for :" << elem.first;
       auto shape = elem.second.first;
-      size_t ssize = runner->GetInputMemSize(elem.first);
+      size_t ssize = runner.GetInputMemSize(elem.first);
       char* data = (char*)malloc(ssize);
       LOG(INFO) << "Random Input Size:" << ssize << "  bytes";
-      runner->SetInput(elem.first, data);
+      runner.SetInput(elem.first, data);
       free(data);
     }
     // Run the model
-    runner->Run();
+    runner.Run();
     // Get Output and dump few values
     for (auto& elem : mInfo.output_info) {
       LOG(INFO) << "Get Output for :" << elem.first;
       auto shape = elem.second.first;
-      size_t ssize = runner->GetOutputMemSize(elem.first);
+      size_t ssize = runner.GetOutputMemSize(elem.first);
       char* data = (char*)malloc(ssize);
-      runner->GetOutput(elem.first, data);
+      runner.GetOutput(elem.first, data);
       LOG(INFO) << "Output Size:" << ssize << "  bytes";
       free(data);
     }
   }
+
+  if (args.profile) {
+    // Print Stats
+    runner.PrintStats();
+  }
   auto tstart = std::chrono::high_resolution_clock::now();
-  delete runner;
+  runner.~TVMRunner();
```

Comment:
Why do you need to call dtor here? Probably, I missed something in the logic.
I suppose that the object will be automatically destroyed in the end of the program.

Reply:
We are timing it to measure the unload time.